cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewManager.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.app.preview;
+
+import co.cask.cdap.api.metrics.MetricTimeSeries;
+import co.cask.cdap.common.NotFoundException;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.id.PreviewId;
+import org.apache.twill.api.logging.LogEntry;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Interface used to start preview and also retrieve the information associated with a preview.
+ */
+public interface PreviewManager {
+
+  /**
+   * Start the preview of an application config provided as an input.
+   * @param namespaceId the id of 
+   * @param config the config for the preview
+   * @return the unique {@link PreviewId} generated for the preview run
+   */
+  PreviewId start(NamespaceId namespaceId, String config) throws Exception;
```

Comment:
Can a preivew run be stop/cancelled? If not, how to terminate a preview run if it takes too long (maybe bug that causes it)?